src/util/windows.jl

```diff
@@ -0,0 +1,66 @@
+"""
+An iterator which moves across a container, as a sliding window.
+"""
+immutable EachWindowIterator{T <: AbstractArray}
```

Comment:
I see what you mean, because you can argue that a string is an ordered array of characters of a given alphabet, and in C this really is the case. Can type unions be used to sort this? I would have any T so long as it inherits from either Sequence, AbstractArray, or AbstractString?